File: 56-merge-intervals/merge-intervals.ts
function merge(intervals: number[][]): number[][] {
    intervals = intervals.sort((a, b) => a[0] - b[0])
    const output = []
    let prev = intervals[0]

    for(let i = 1; i < intervals.length; i++) {
        if(prev[1] < intervals[i][0]) {
            output.push(prev)
            prev = intervals[i]
        }  else {
            prev[1] = Math.max(prev[1], intervals[i][1])
        }
    }

    output.push(prev)
    return output
};

File: 56-merge-intervals/merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        if len(intervals) < 2:
            return intervals
        arr = []
        intervals.sort(key = lambda x : x[0])

        for i in range(1, len(intervals)):
            if intervals[i -1][1] < intervals[i][0]:
                arr.append(intervals[i - 1])
            else:
                intervals[i] = [
                    min(intervals[i - 1][0], intervals[i][0]),
                    max(intervals[i - 1][1], intervals[i][1])
                ]

        arr.append(intervals[-1])
        
        return arr